osu.Game/Overlays/Mods/ModColumn.cs

```diff
@@ -206,8 +210,13 @@ public void DeselectAll()
         {
             pendingSelectionOperations.Clear();
 
-            foreach (var button in availableMods.Where(b => b.Active.Value && !b.Filtered.Value))
-                pendingSelectionOperations.Enqueue(() => button.Active.Value = false);
+            foreach (var button in availableMods.Where(b => b.Active.Value))
+            {
+                if (Alpha == 0f)
+                    button.Active.Value = false; //If column is hidden change state manually without any animation
```

Comment:
Doing this is going to make https://github.com/ppy/osu/issues/20091 actually unbearable.

Reply:
This line is necessary. Without it `DeselectAllModsButton` will deselect all **visible** mods, which is wrong imo. Would adding code below to `ModSelectPanel` be a compromise?
```diff
  private void playStateChangeSamples()
  {
      if (samplePlaybackDisabled.Value)
          return;
+
+     if (!IsPresent)
+         return;

      if (Active.Value)
          sampleOn?.Play();
      else
          sampleOff?.Play();
  }
```